tests/0113-cooperative_rebalance.cpp

```diff
@@ -26,227 +26,339 @@
  * POSSIBILITY OF SUCH DAMAGE.
```

Comment:
There is alot of code repitition which makes the code hard to read, could really do with some helper functions to check assignments, call subscribe, etc, just some sane abstraction so that the test-cases themselves describe the test logic rather than itty nitty API details.

Reply:
yep totally agree, i didn't give it any thought. it's easy to make it much better.